src/Microsoft.SqlTools.ManagedBatchParser/ReliableConnection/ReliableSqlConnection.cs

```diff
@@ -63,6 +63,9 @@ public sealed partial class ReliableSqlConnection : DbConnection, IDisposable
         public ReliableSqlConnection(string connectionString, RetryPolicy connectionRetryPolicy, RetryPolicy commandRetryPolicy, string azureAccountToken)
         {
             _underlyingConnection = new SqlConnection(connectionString);
+
+            _underlyingConnection.RetryLogicProvider = RetryPolicyUtils.ServerlessWaitRetryLogicProvider();
```

Comment:
I wonder if someone would want to customize this retry logic provider applying on SqlConnection.

@Matteo-T any thoughts? Since it affects Managed Batch parser, what is your thought in specifying this provider by default. It seems to me it's a necessity to have a retry logic provider for serverless connections, but would you need to customize the retry attempt and intervals?

Reply:
@cheenamalhotra - I cannot really tell the impact of this Provider overall (it's probably negligible for the way I use it), but... the idea of being able to either customize this "retry provider" (as I wrote elsewhere, seeing a "Serverless" in the name confuses me), reset to "null" (=no provider, which seems to be the default in MDS), or even pass my own would be kind of cool. Maybe it's not a lot of work to make it more general-purpose without over-engineering it?